src/components/vm/snapshots/vmSnapshotsCreateModal.jsx

```diff
@@ -139,27 +190,37 @@ export class CreateSnapshotModal extends React.Component {
         else if (!name)
             validationError.name = _("Name can not be empty");
 
-        if (isExternal && vm.state === "running" && !memoryPath)
-            validationError.memory = _("Memory file can not be empty");
+        if (isExternal && vm.state === "running" && !memoryLocation)
+            validationError.memory = _("Memory save location can not be empty");
 
         return validationError;
     }
 
     onCreate() {
         const Dialogs = this.context;
         const { vm, isExternal } = this.props;
-        const { name, description, memoryPath } = this.state;
+        const { name, description, memoryLocation } = this.state;
         const validationError = this.onValidate();
 
         if (!Object.keys(validationError).length) {
             this.setState({ inProgress: true });
-            snapshotCreate({
-                vm,
-                name,
-                description,
-                isExternal,
-                memoryPath: isExternal && vm.state === "running" && memoryPath,
-            })
+            let mpath = null;
+            if (isExternal && vm.state === "running" && memoryLocation) {
+                mpath = memoryLocation;
+                if (mpath[mpath.length - 1] != "/")
+                    mpath = mpath + "/";
+                mpath = mpath + vm.name + "." + name + ".save";
+            }
+            const superuser = (vm.connectionName === LIBVIRT_SYSTEM_CONNECTION);
+            cockpit.spawn(["mkdir", "-p", memoryLocation], { superuser, err: "message" })
```

Comment:
Technically `superuser` is not a boolean option but it 'try' or 'require' https://cockpit-project.org/guide/latest/cockpit-spawn.html

I think cockpit-ws still has code to convert true into "require" but depending on that fact feels wrong.

Reply:
Ok! I wasn't sure which one is the deprecated way, and which one isn't. Thanks for clearing that up!